ros2snap

```diff
@@ -1,112 +1,345 @@
+      print "Parsing metadata from package.xml and writing to meta/package.yaml"
+      self.parse_write_metadata()
+      self.copy_env_scripts()
+
+    if copy_recursive_deps:
+      print "Copying all recursive run dependencies into snap"
+      self.copy_recursive_dependencies(self.package)
+      for key in core_rosdeps:
+        self.resolve_and_copy(key)
+
+    self.cache.close()
+
+    if build_snap:
+      os.system("snappy build snappy_build/" + self.package_key_final)
+
+def prepare_from_source(package_key):
+  path = os.getcwd()
+
+  packages = catkin_pkg.packages.find_packages(path + "/src/")
+
+  if len(packages) == 0:
+    print "No packages found in catkin workspace. Exiting."
+    sys.exit(-1)
+
+  os.system("catkin_make install")
+
+  builder = SnappyBuilder(package_key, packages, "install/")
+
+  builder.create_dir_structure()
+  # Copy all files in install to snappy_build/<package>
+  # If building from source, need to also get the setup files
+  check_create_dir(builder.snappy_root + "install/")
+
+  for path in os.listdir(builder.pkg_root):
+    if os.path.isfile(builder.pkg_root + "/" + path):
+      shutil.copy2(builder.pkg_root + "/" + path, \
+          builder.snappy_root + "install/")
+
+  builder.copy_files_from_pkg_root()
+  return builder
+
+def prepare_from_debs(package_key):
+  # TODO: Install the package debian if it is not found
+
+  ros_path = os.path.dirname(os.getenv("ROS_ROOT"))
+  packages = catkin_pkg.packages.find_packages(ros_path)
+
+  if len(packages) <= 0:
+    print "Error: no catkin packages found in " + ros_path + ". Exiting."
+    sys.exit(-1)
+
+  builder = SnappyBuilder(package_key, packages, os.path.dirname(ros_path) + "/")
+
+  builder.create_dir_structure()
+  builder.copy_files_from_pkg_root()
+  # get all the stuff we need that isn't under the package name
+  key_entry = bloom.generators.common.resolve_rosdep_key(builder.package_key, "ubuntu", "trusty")
+  if key_entry is None or (key_entry[0] is None) or (len(key_entry[0]) == 0):
+    print "Apt couldn't find package with name " + package_key + ". Exiting."
+    sys.exit(-1)
+
+  apt_name = key_entry[0][0]
+  pkg = builder.cache[apt_name]
+  builder.copy_files(pkg.installed_files)
+  return builder
+
+
+def main():
+  parser = argparse.ArgumentParser(description="Build a Snappy package (snap) from a ROS package.")
+  parser.add_argument('package', help="Name of the ROS package to snap.")
+  parser.add_argument('--debs', '-d', help="If true, build ROS package from debians. Else, build from catkin workspace rooted at current dir.", default=0)
+  parser.add_argument('--metadata_generate', '-m', help="Enable/disable generating Snappy metadata for this package.", default=1)
+  parser.add_argument('--copy_dependencies', '-c', help="Enable/disable copying all recursive dependencies for this package.", default=1)
+  parser.add_argument('--snap_build', '-s', help="Enable/disable calling 'snappy_build' for this package.", default=1)
```

Comment:
Instead of using the `bloom` API directly use the `rosdep` API. The bloom function invoked is https://github.com/ros-infrastructure/bloom/blob/8f58cae7cb8d103799322d5fe5eecaee322ebaf3/bloom/generators/common.py#L120-L160 

Instead you could create a very stripped down version of that function and use that instead (which will then only rely on `rosdep` and not on `bloom`).